Events.php

```diff
@@ -34,11 +34,14 @@ public static function onAdminMenuInit($event)
     }
 
     public static function addAdFrame($event)
-    {
-        if (Yii::$app->user->isGuest || Yii::$app->user->identity && AuthHelper::isGuestAccessEnabled())
+    {	 
+        if (
+			Yii::$app->user->isGuest && 
+			AuthHelper::isGuestAccessEnabled() &&
+			(!Yii::$app->getModule('legal') || (isset($_COOKIE['cookieconsent_status']) && 'dismiss' === $_COOKIE['cookieconsent_status'] ) )
```

Comment:
I agree here that the Legal module should be respected when it comes to cookie consent is requested, but here are some questions that need to be answered in regards to such a feature;

1. Is it checked to make sure that the Legal module is installed or enabled?
2. If not installed or enabled does this throw any errors?

Reply:
Hi,
please excuse my late response.

> From reviewing the code itself, I see that this will only display ads to guests and not those that are logged in, am I >correct in this?

Yes, this is correct!
So far it is an requirement of my own implementation, further if you show ads on pages the google bot cannot spider, google may "meckern" saying you do not comply to their terms.
As I know, Google offers the possibility to give their bots special access (passwords) to crawl these protected sites, for me personally this is not an option and I do not plan to implement this in the near future.

>Is it checked to make sure that the Legal module is installed or enabled?
>If not installed or enabled does this throw any errors?

No, it just checks if the module is ~enabled~ present, if it is enabled it requires the consent cookie to be correct, if it is not present just ignore the cookie check.

I am very new to HumHub coding, so as I guess Yii::$app->getModule('legal') only checks if the module is present, not if it is enabled? Am I correct?

reagrds,
Till

Reply:
Maybe an Event Handler would be a better solution here, I'd like to ask for some outside insight on this @luke-, what are your thoughts about this?